src/test/java/com/google/javascript/cl2dts/shouldResolveForwardDeclares/forward.js

```diff
@@ -0,0 +1,11 @@
+goog.provide('forward.A');
```

Comment:
So turns out that is not needed for closure to consider the type as present when going through forward.js even if it hasn't seen `forward.D`. I am guess that is a feature of closure?

Then it begs the question what is `goog.forwardDeclare` needed for?